### PR TITLE
feat: add config api

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
# Description

## Issue reference

Please reference the issue this PR will close: #2988

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by yaron2** [comment]
@LaurenceLiZhixin notice the checks are failing

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -1071,6 +1083,31 @@ func (a *DaprRuntime) initOutputBinding(c components_v1alpha1.Component) error {
 	return nil
 }
 
+func (a *DaprRuntime) initConfiguration(s components_v1alpha1.Component) error {
+	store, err := a.configurationStoreRegistry.Create(s.Spec.Type, s.Spec.Version)
```
[the remark this answers is not in the document]
Comment (yaron2) [reply pkg/runtime/runtime.go:1106]: @LaurenceLiZhixin please look at the State API to version the proto method to specify this is Alpha. So basically add `Alpha1` to the methods like here: https://github.com/dapr/dapr/pull/3695/files#diff-da511bac0105b1e517ff78c6351b68bda0d5862cb0daae89127f6c0250551265R34.

**Comment by daixiang0** [comment]
@LaurenceLiZhixin still fail.

**Comment by LaurenceLiZhixin** [comment]
> @LaurenceLiZhixin still fail.

Could you please restart the e2e test again? I guess this problem is not my fault that caused it. And the previous commit can pass all tests.

File: pkg/runtime/runtime.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (daixiang0) [reply pkg/runtime/runtime.go:1106]: @LaurenceLiZhixin please fix all unresloved comments.

**Comment by daixiang0** [comment]
@LaurenceLiZhixin Please ensure resolve all comments.

File: cmd/daprd/main.go [hunk 2]
```diff
@@ -237,6 +241,11 @@ func main() {
 				return state_mysql.NewMySQLStateStore(logContrib)
 			}),
 		),
+		runtime.WithConfigurations(
+			configuration_loader.New("redis.native", func() configuration.Store {
```
[the remark this answers is not in the document]
Comment (LaurenceLiZhixin) [reply cmd/daprd/main.go]: Done

File: pkg/runtime/runtime.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (LaurenceLiZhixin) [reply pkg/runtime/runtime.go:1106]: Done @yaron2 @daixiang0

**Comment by LaurenceLiZhixin** [comment]
> @LaurenceLiZhixin Please ensure resolve all comments.

Done

**Comment by yaron2** [comment]
Thanks @LaurenceLiZhixin resolve conflicts please.

**Comment by seeflood** [comment]
Cool!

**Review by artursouza — COMMENTED** [review]
LGTM

**Pull request merged**